src/tasks/MonoTargetsTasks/ILStrip/ILStrip.cs

```diff
@@ -81,4 +122,124 @@ private bool StripAssembly(ITaskItem assemblyItem)
+            return true;
+        }
+
+        using (StreamReader sr = new StreamReader(methodTokenFile))
+        {
+            string? assemblyFilePath = sr.ReadLine();
+            if (string.IsNullOrEmpty(assemblyFilePath))
+            {
+                return true;
+            }
+
+            if (!File.Exists(assemblyFilePath))
+            {
+                Log.LogMessage(MessageImportance.Low, $"[ILStrip] {assemblyFilePath} doesn't exit.");
+                return true;
+            }
+
+            string? line = sr.ReadLine();
+            if (!string.IsNullOrEmpty(line))
+            {
+                string? assemblyPath = Path.GetDirectoryName(assemblyFilePath);
+                string? assemblyName = Path.GetFileNameWithoutExtension(assemblyFilePath);
+                string trimmedAssemblyFilePath;
+                string newName_assemblyFilePath;
+                if (string.IsNullOrEmpty(assemblyPath))
+                {
+                    trimmedAssemblyFilePath = assemblyName + "_new.dll";
+                    newName_assemblyFilePath = assemblyName + "_old.dll";
+                }
+                else
+                {
+                    trimmedAssemblyFilePath = Path.Combine(assemblyPath, (assemblyName + "_new.dll"));
+                    newName_assemblyFilePath = Path.Combine(assemblyPath, (assemblyName + "_old.dll"));
+                }
+
+                using (FileStream fs = File.Open(assemblyFilePath, FileMode.Open),
+                    os = File.Open(trimmedAssemblyFilePath, FileMode.Create))
+                {
+                    MemoryStream memStream = new MemoryStream((int)fs.Length);
+                    fs.CopyTo(memStream);
+
+                    fs.Position = 0;
+                    PEReader peReader = new PEReader(fs, PEStreamOptions.LeaveOpen);
+                    MetadataReader mr = peReader.GetMetadataReader();
+
+                    Dictionary<int, int> token_to_rva = new Dictionary<int, int>();
+                    Dictionary<int, int> method_body_uses = new Dictionary<int, int>();
+
+                    foreach (MethodDefinitionHandle mdefh in mr.MethodDefinitions)
+                    {
+                        int methodToken = MetadataTokens.GetToken(mr, mdefh);
+                        MethodDefinition mdef = mr.GetMethodDefinition(mdefh);
+                        int rva = mdef.RelativeVirtualAddress;
+
+                        token_to_rva.Add(methodToken, rva);
+
+                        if (method_body_uses.TryGetValue(rva, out var count))
+                        {
+                            method_body_uses[rva]++;
+                        }
+                        else
+                        {
+                            method_body_uses.Add(rva, 1);
+                        }
+                    }
+
+                    do
+                    {
+                        int methodToken2Trim = Convert.ToInt32(line);
```

Comment:
Do this to read `0x0900abcd` hex token values:

```suggestion
                        int methodToken2Trim = Convert.ToInt32(line, 16);
```

Reply:
Fixed.